src/components/navbar.jsx

```diff
@@ -16,9 +17,16 @@ function NavbarComponent() {
                         </Nav.Link>
                     </Nav>
                     <Nav>
-                        <Nav.Link as={Link} to="/login">
-                            Login
-                        </Nav.Link>
+                        {/* kako napraviti da se ovo osvezi kada se state promeni? */}
+                        {!isTokenValid() ? (
```

Comment:
Moras koristiti state direktno da bi imao reagovanje na promene u state-u.  Zato bi mnogo bilo lakse da si radio sa selectorima kao sto sam i spominjao vec da je bolje da radis. Jer onda preko selektora dovlacis nesto iz state-a direktno, npr u tvom slucaju token bi mogao da dovuces.

Dakle, umesto sto u user-utils proveravas da li je token validan, moraces to bas u komponentama da radis da bi koristio `useSelector`.
Nesto ovako bi imao:
```
const currentUser = useSelector(state => state.user.current);
```
Arrow funkcija `state => state.user.current` bi generalno bilo bolje da bude izdvojeno na jedno mesto (u slucaju ponavljanja). To isto mozes videti kako sam ja radio u onom todo repo-u.

Dakle tako bi uzeo current usera u komponenti i onda bi umesto `!isTokenValid()...` imao `isExpired(currentUser.token)...`